src/game/reducers/update-discard-pile/index.ts

```diff
@@ -0,0 +1,18 @@
+import { IGame, IPlayer } from '../../types'
+
+export const updateDiscardPile = (
+  game: IGame,
+  playerId: IPlayer['id'],
+  discardPile: IPlayer['discardPile']
+): IGame => {
+  return {
+    ...game,
+    table: {
+      ...game.table,
+      players: {
+        ...game.table.players,
+        ...{ [playerId]: { ...game.table.players[playerId], discardPile } },
+      },
+    },
+  }
```

Comment:
this is purely an opinion comment so feel free to ignore, but all this destructuring is a lot to parse mentally. it would be a lot simpler if it was something like this:

```js
const newGame = { ...game }
newGame.table.players[playerId].discardPile = { ...discardPile }

return newGame
```

Reply:
That's a great point. The idea here was to avoid mutating state and respecting the `readonly` attributes of the game properties. I've adapted your suggestion for the branch based off of this one: https://github.com/jeremyckahn/farmhand-shuffle/commit/f9c147717492a6a5c3f00b5e249d3ac30e9d6100